### DISCUSSION TITLE
How to create a new control: SearchBox.

### DISCUSSION THREAD

**Discussion by Nemirtingas** [AvaloniaUI/Avalonia#5889] [Q&A] [answered]
Hi,

I'd like to overload the combobox control to use it as a searchbox. Basically, it would just be a TextBox with a popup menu that can be filled by a custom function/list binding to send back to the user something that could match what they are typing.

The SearchBox could be filled with any input the user types like a TextBox and the search hints should not be mandatory, the user is free to use the results or not. In the ComboBox, the user need to pick a result.

![image](https://user-images.githubusercontent.com/9432037/117566435-ebb55c80-b0b6-11eb-9b39-57bff393b140.png)

I got some kind of working SearchBox, its a TextBox + Popup menu that I open when the user types something but I have issue making the popup "follow" the TextBox.

**Comment by FoggyFinder** [comment, marked as answer]
There is [AutoCompleteBox](http://reference.avaloniaui.net/api/Avalonia.Controls/AutoCompleteBox/) already

**Comment by Nemirtingas** [comment]
Ok thanks, right now I don't understand how it works. I was triggering a search on text change and filling the popup asynchronously, seems different in the AutoCompleteBox. When typing in it, the Text property isn't notified in my code.
I'll search in the AutoCompleteBoxPage.xaml sample.

**Comment by Nemirtingas** [comment]
![image](https://user-images.githubusercontent.com/9432037/117567902-93825880-b0be-11eb-9f48-2c4eaae9b8f3.png)

AutoCompleteBox seems to bug like mine ;)
